Jenkinsfile

```diff
@@ -1,90 +1,118 @@
 #!groovy
 import groovy.json.JsonOutput
+import groovy.json.JsonBuilder
 import groovy.json.JsonSlurper
 
 node {
-	stage('Read test names') {
-        checkout scm
-	    stash name: 'scripts', includes: 'um2junit.rb'
+    stage('Read and Execute tests') {
         // Obtain credentials for accessing the useMango server, which should be stored in Jenkins with the ID of 'usemango'
-	    withCredentials([usernamePassword(credentialsId: 'usemango', usernameVariable: 'user', passwordVariable: 'pwd')]) {
+        withCredentials([usernamePassword(credentialsId: '', usernameVariable: 'user', passwordVariable: 'pwd')]) {
             // Obtain values for the server url, project name and folder name from a Jenkins config file, which have the ID
             // that matches the name of the current pipeline job.
-            configFileProvider([configFile(fileId: env.JOB_NAME, targetLocation: 'test.props')]) {
-                def props = readProperties file: 'test.props'
-                echo "Running tests for server ${props.server}, project ${props.project}, folder ${props.folder}"
-                def credentials = [Email:user,Password:pwd,ExecutionOnly:true]
-                String cookie = getAuthenticationCookie(props.server, credentials)
-                def tests = getTests(props.server, props.project, props.folder, cookie)
-                def testJobs = [:]
-                for (testname in tests) {
-                    def tn = testname
-                    echo "Scheduling ${tn}"
-                    testJobs[tn] = {
-                        node('usemango') {
-                            try {
-                                unstash 'scripts'
-                                bat "\"%programfiles(x86)%\\Infuse Consulting\\useMango\\App\\MangoMotor.exe\" -s ${props.server} -p ${props.project} --testname \"${tn}\" -e ${user} -a ${pwd}"
-                            }
-                            finally {                                
-                                junit \"%PROGRAMDATA%\\useMango\\logs\\junit.xml\"
+            String TEST_SERVICE_URL = "https://e2zkwyufoa.execute-api.eu-west-1.amazonaws.com/v1"
+            String SCRIPTS_SERVICE_URL = "https://9viq7bzai3.execute-api.eu-west-1.amazonaws.com/v1"
+            String APP_WEBSITE_URL = "https://app.dev.usemango.co.uk"     
+            echo "Running tests in project ${params['Project']} with tags ${params['Tags']}"
+            def tests = getTests(TEST_SERVICE_URL)
+            def testJobs = [:]
+            def testResults = [:]
+            tests.eachWithIndex { id, index ->
+                echo "Scheduling ${id}"
+                    testJobs[id] = {
+                        node('') {
+                            wrap([$class: "MaskPasswordsBuildWrapper", varPasswordPairs: [[password: ""]]]) {
+                                dir ("${env.WORKSPACE}\\${tests[index]}") {
```

Comment:
We're duplicating this `dir` function a few time, please have a look if this can be re-factored.